cowprotocol/cowswap
# Add claim vesting iframe hook

# Summary

- Create the Claim Vesting Hook in the dapps folder as an IFrame linked to Bleu vercel deployed app
- Add Claim Vesting Hook in the hookStore, as both pre and post hook, in all chains the LlamaPay vesting factory is implemented (Mainnet, Gnosis and Arbitrum)
- Edit IFrameContainer min-height to 300px for aesthetic reasons

![image](https://github.com/user-attachments/assets/c12d73ee-2814-4e24-a0c8-ede26bc960ba)

![image](https://github.com/user-attachments/assets/963a5ec2-c6a4-49e0-8a50-fe0e13ac5cdb)

![image](https://github.com/user-attachments/assets/ee4c77e3-9881-46d0-b1bd-495c3e17290a)

![image](https://github.com/user-attachments/assets/c8b69a40-dd02-4bb7-87c0-05f78e2acb16)

Errors cases:
- Insert a valid Ethereum address
- Please connect your wallet first
- Address is not a valid vesting contract
- You are not the contract recipient

Example:
![image](https://github.com/user-attachments/assets/1e7f540e-c097-480a-afba-7dec0706117d)

Success case:
![image](https://github.com/user-attachments/assets/70c94e86-4613-4a9f-a241-31b262fb077e)

# How to test:

1. Access /#/100/swap/hooks/WXDAI

2. Connect your wallet and switch to Gnosis chain

3. In Hooks > Add Pre-Hook Action, the Claim Vesting Hook should appear in the "All hooks" section

4. Insert a valid vesting contract. Here is a mapping of addresses to vesting contracts that should work:

Address:
0x9FA3c00a92Ec5f96B1Ad2527ab41B3932EFEDa58
Vesting Contract:
0x77e62665422fc216ca278b7655b3329dc5ba173c

Address:
0xFF714b8b0e2700303eC912BD40496C3997ceEa2b
Vesting Contract:
0xf078d4ADBa99BaC7Ab4a64ea3F24b6307107198d

Address:
0x8FAb71C0d4272698A3B2d1F3Ed5FC3c1B9b3E531
Vesting Contract:
0xac7CBD5DD1D64bF6C9cdCceCf0cb7c909C3AF4A6
